### ISSUE TITLE
Very cool!

### ISSUE THREAD

**Issue by viseshrp** [basnijholt/markdown-code-runner#13]
Cool tool! Thank you for your work. You might want to checkout [Cog](https://github.com/nedbat/cog)  for inspiration. That's all I had. Thank you again!

**Comment by basnijholt** [comment]
Hi @viseshrp, thanks for your interest!

Some differences:

[`markdown-code-runner`](https://github.com/basnijholt/markdown-code-runner):

*   Designed for Markdown files, works with native code blocks.
*   Executes Python and Bash code, updates output in-place.
*   Supports hidden code blocks for seamless content generation.
*   Lightweight, easy GitHub Actions integration.

Cog:

*   General-purpose code generation for any text file.
*   Embeds Python code in source files using special comments.
*   Doesn't support native Markdown code blocks or hiding code.
*   Requires a separate step to execute code and update output.

In summary, [`markdown-code-runner`](https://github.com/basnijholt/markdown-code-runner) is tailored for Markdown with support for **native code blocks and hidden code**, while Cog is more versatile but doesn't support these Markdown-specific features. Choose the tool that best fits your needs.

**Comment by pawamoy** [comment]
Shameless plug :sweat_smile: [Markdown Exec](https://pawamoy.github.io/markdown-exec/). It's a Python-Markdown extension so even more integrated with the rendering process, *but* it only works with Python-Markdown and the PyMDown-Extensions' SuperFences extension (so it's less generic).

Very nice project and README! Nice work :slightly_smiling_face:

**Comment by basnijholt** [comment]
Awesome!

I will add a section referring to other tools :)
